`tt_metal/api/tt-metalium/distributed_context.hpp` needs to include `<cstdint>`.

### Describe the bug

After #21870 I see a build failure on our side. It's probably because of slightly different version of C++ library:
```
In file included from <...>/tt-metal/tt_metal/distributed/mpi_distributed_context.cpp:6:
In file included from <...>/tt-metal/tt_metal/distributed/mpi_distributed_context.hpp:9:
<...>/tt-metal/tt_metal/api/tt-metalium/distributed_context.hpp:16:23: error: no type named 'uint8_t' in namespace 'std'; did you mean simply 'uint8_t'?
   16 | enum class ReduceOp : std::uint8_t { SUM, MAX, MIN, PROD, LAND, LOR, BAND, BOR };
      |                       ^~~~~~~~~~~~
      |                       uint8_t
/usr/include/x86_64-linux-gnu/bits/stdint-uintn.h:24:19: note: 'uint8_t' declared here
   24 | typedef __uint8_t uint8_t;
      |                   ^
```

`uint8_t` is defined in `<cstdint> and it's better to have it explicitly included.

### Steps to reproduce the issue

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error


### Expected behavior

_No response_

### Please complete the following environment information

_No response_

--- a/tt_metal/api/tt-metalium/distributed_context.hpp
+++ b/tt_metal/api/tt-metalium/distributed_context.hpp
@@ -9,6 +9,7 @@
 #include <tt_stl/span.hpp>
 #include <optional>
 #include <cstddef>
+#include <cstdint>
 #include <complex>
 
 namespace tt::tt_metal::distributed::multihost {

